core/storage/user/gae_models.py

```diff
@@ -850,6 +997,18 @@ def get_deletion_policy():
         """
         return base_models.DELETION_POLICY.DELETE
 
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check that CollectionProgressModel exist for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether the models for user_id exists.
+        """
+        return cls.get_all().filter(cls.user_id == user_id).count() > 0
```

Comment:
Here and elsewhere: would it be more efficient to use get() to return the first result, rather than doing the full count? If there are a lot of entities, counting might take more time.